views/js/controller/creator/helpers/baseType.js

```diff
@@ -0,0 +1,333 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * The BaseType enumeration (port of \qtism\common\enums\BaseType).
+ *
+ * From IMS QTI:
+ *
+ * A base-type is simply a description of a set of atomic values (atomic to this specification).
+ * Note that several of the baseTypes used to define the runtime data model have identical
+ * definitions to those of the basic data types used to define the values for attributes
+ * in the specification itself. The use of an enumeration to define the set of baseTypes
+ * used in the runtime model, as opposed to the use of classes with similar names, is
+ * designed to help distinguish between these two distinct levels of modelling.
+ *
+ * @author Jérôme Bogaerts <jerome@taotesting.com>
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+define([
+    'lodash'
+], function (_) {
```

Comment:
The reason why is because I only found numerical values in the JSON provided by the server.
For the purpose of this feature, a list of simple constants could do the job, but I integrated the full helper to provide more support. If you think this is overkill I can replace it.

For the cardinalities, I did not see it, just wondering the numbers was the allowed occurrences numbers. So I will replace the magic numbers by constants.